webrender/src/device/gfx.rs

```diff
@@ -3186,7 +3186,7 @@ impl<B: hal::Backend> Device<B> {
     pub fn free_image(&mut self, texture: &mut Texture) {
         // Note: this is a very rare case, but if it becomes a problem
         // we need to handle this in renderer.rs
-        if texture.still_in_flight(self.frame_id, self.frame_count) {
+        if texture.still_in_flight(self.frame_id, self.frame_count) && !self.upload_queue.is_empty() {
```

Comment:
So there is a note, which I think got resolved with [this](https://github.com/szeged/webrender/pull/227) pr.
This is a special case, because the still_in_flight will always be true if our frame_count is higher that 1. But actually we won't use this anymore since we are freeing all the resources.
So the only thing you should have in this if check is
```
if texture.still_in_flight(self.frame_id, self.frame_count) {
    self.wait_for_resources();
}
```
And the case you try to solve with this (freeing images with deinit), you already waited for the resources so this won't do anything special.